tests/std/tests/P0009R18_mdspan_mdspan/test.cpp

```diff
@@ -0,0 +1,1352 @@
+        static_assert(is_constructible_v<Mds, double*, array<ConvertibleToInt<ptrdiff_t, IsNothrow::yes>, 2>>);
+        static_assert(!is_constructible_v<Mds, double*, span<ConvertibleToInt<ptrdiff_t, IsNothrow::no>, 2>>);
+        static_assert(!is_constructible_v<Mds, double*, array<ConvertibleToInt<ptrdiff_t, IsNothrow::no>, 2>>);
+    }
+
+    { // Check constraint: 'N == rank() || N == rank_dynamic()'
+        using Mds = mdspan<int*, extents<short, 2, 3, dynamic_extent, dynamic_extent>>;
+        static_assert(!is_constructible_v<Mds, int**, span<int, 1>>);
+        static_assert(!is_constructible_v<Mds, int**, array<int, 1>>);
+        static_assert(is_constructible_v<Mds, int**, span<int, 2>>);
+        static_assert(is_constructible_v<Mds, int**, array<int, 2>>);
+        static_assert(!is_constructible_v<Mds, int**, span<int, 3>>);
+        static_assert(!is_constructible_v<Mds, int**, array<int, 3>>);
+        static_assert(is_constructible_v<Mds, int**, span<int, 4>>);
+        static_assert(is_constructible_v<Mds, int**, array<int, 4>>);
+    }
+
+    { // Check constraint: 'is_constructible_v<mapping_type, extents_type>'
+        static_assert(is_constructible_v<mdspan<const int, dextents<int, 2>, layout_left>, int* const, span<int, 2>>);
+        static_assert(is_constructible_v<mdspan<const int, dextents<int, 2>, layout_left>, int* const, array<int, 2>>);
+        static_assert(
+            is_constructible_v<mdspan<const int, dextents<int, 2>, TrackingLayout<layout_left, RequireId::no>>,
+                int* const, span<int, 2>>);
+        static_assert(
+            is_constructible_v<mdspan<const int, dextents<int, 2>, TrackingLayout<layout_left, RequireId::no>>,
+                int* const, array<int, 2>>);
+        static_assert(!is_constructible_v<mdspan<int, dextents<int, 2>, layout_stride>, int*, span<int, 2>>);
+        static_assert(!is_constructible_v<mdspan<int, dextents<int, 2>, layout_stride>, int*, array<int, 2>>);
+        static_assert(!is_constructible_v<mdspan<int, dextents<int, 2>, TrackingLayout<>>, int*, span<int, 2>>);
+        static_assert(!is_constructible_v<mdspan<int, dextents<int, 2>, TrackingLayout<>>, int*, array<int, 2>>);
+    }
+
+    { // Check constraint: 'is_default_constructible_v<accessor_type>'
+        static_assert(is_constructible_v<mdspan<bool, dextents<int, 2>, layout_right, VectorBoolAccessor>,
+            vector<bool>::iterator, span<short, 2>>);
+        static_assert(is_constructible_v<mdspan<bool, dextents<int, 2>, layout_right, VectorBoolAccessor>,
+            vector<bool>::iterator, array<short, 2>>);
+        static_assert(!is_constructible_v<mdspan<int, dextents<int, 2>, layout_right, TrackingAccessor<int>>, int*,
+                      array<int, 2>>);
+        static_assert(!is_constructible_v<mdspan<int, dextents<int, 2>, layout_right, TrackingAccessor<int>>, int*,
+                      span<int, 2>>);

```

Comment:
Style nitpick: Everything else has said `span`, then `array`, but this pair is using the other order.